mem/ctrl/ContextSwitchRTL.py

```diff
@@ -0,0 +1,95 @@
+"""
+==========================================================================
+ContextSwitchRTL.py
+==========================================================================
+Records/resumes progress (itertion) for functional unit (FU).
+
+Author : Yufei Yang
+  Date : Aug 11, 2025
+"""
+from pymtl3.stdlib.primitive import *
+from ...lib.basic.val_rdy.ifcs import ValRdyRecvIfcRTL as RecvIfcRTL
+from ...lib.basic.val_rdy.ifcs import ValRdySendIfcRTL as SendIfcRTL
+from ...lib.cmd_type import *
+from ...lib.opt_type import *
+from ...lib.util.common import *
+
+class ContextSwitchRTL(Component):
+
+  def construct(s, CgraPayloadType, DataType, CtrlType, num_status = 3):
+
+    # Constant
+    StatusType = mk_bits(clog2(num_status))
+
+    # Interface
+    s.recv_cmd = RecvIfcRTL(CgraPayloadType)
+    s.recv_opt = RecvIfcRTL(CtrlType)
+    s.progress_in = RecvIfcRTL(DataType)
+    s.progress_out = SendIfcRTL(DataType)
```

Comment:
For these three:
```
    s.recv_opt = RecvIfcRTL(CtrlType)
    s.progress_in = RecvIfcRTL(DataType)
    s.progress_out = SendIfcRTL(DataType)
```
Should we just use `InPort` or `RecvIfc`? I think `InPort` might be easier so we don't need to check val/rdy?